File: app/src/main/java/com/hjc/reader/ui/gank/adapter/FilterAdapter.java
package com.hjc.reader.ui.gank.adapter;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.hjc.reader.R;
import com.hjc.reader.databinding.ItemFilterBinding;


import java.util.List;

public class FilterAdapter extends BaseQuickAdapter<String, BaseViewHolder> {

    public FilterAdapter(@Nullable List<String> data) {
        super(R.layout.item_filter, data);
    }

    @Override
    protected void onItemViewHolderCreated(@NonNull BaseViewHolder viewHolder, int viewType) {
        DataBindingUtil.bind(viewHolder.itemView);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, String item) {
        if (item == null) {
            return;
        }

        int imageResId;
        switch (item) {
            case "全部":
                imageResId = R.mipmap.icon_filter_all;
                break;

            case "Android":
                imageResId = R.mipmap.icon_filter_android;
                break;

            case "IOS":
                imageResId = R.mipmap.icon_filter_ios;
                break;

            case "App":
                imageResId = R.mipmap.icon_filter_app;
                break;

            case "前端":
                imageResId = R.mipmap.icon_filter_web;
                break;

            case "后端":
                imageResId = R.mipmap.icon_filter_video;
                break;

            case "推荐":
                imageResId = R.mipmap.icon_filter_extra;
                break;

            default:
                imageResId = R.mipmap.icon_filter_all;
                break;
        }

        ItemFilterBinding binding = helper.getBinding();
        if (binding != null) {
            binding.setTitle(item);
            binding.ivPic.setImageResource(imageResId);
        }
    }
}

File: module-gank/src/main/java/com/hjc/learn/gank/adapter/FilterAdapter.kt
package com.hjc.learn.gank.adapter

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.hjc.learn.gank.R
import com.hjc.learn.gank.databinding.GankItemFilterBinding

/**
 * @Author: HJC
 * @Date: 2021/2/23 16:39
 * @Description: 分类adapter
 */
class FilterAdapter(data: MutableList<String>?) :
    BaseQuickAdapter<String, BaseViewHolder>(R.layout.gank_item_filter, data) {

    override fun onItemViewHolderCreated(viewHolder: BaseViewHolder, viewType: Int) {
        DataBindingUtil.bind<ViewDataBinding>(viewHolder.itemView)
    }

    override fun convert(holder: BaseViewHolder, item: String) {
        val imageResId = when (item) {
            "全部" -> R.mipmap.gank_icon_filter_all
            "Android" -> R.mipmap.gank_icon_filter_android
            "IOS" -> R.mipmap.gank_icon_filter_ios
            "App" -> R.mipmap.gank_icon_filter_app
            "前端" -> R.mipmap.gank_icon_filter_web
            "后端" -> R.mipmap.gank_icon_filter_video
            "推荐" -> R.mipmap.gank_icon_filter_extra
            else -> R.mipmap.gank_icon_filter_all
        }

        val binding = DataBindingUtil.getBinding<GankItemFilterBinding>(holder.itemView)
        binding?.let {
            it.title = item
            it.ivPic.setImageResource(imageResId)
        }

    }
}